src/index.ts

```diff
@@ -1,20 +1,35 @@
 import { useCallback, useLayoutEffect, useReducer, useRef } from 'react'
 import shallowEqual from './shallowEqual'
 
-type StateListener<T> = (state: T) => void
-type StateSelector<T, U> = (state: T) => U
-type PartialState<T> = Partial<T> | ((state: T) => Partial<T>)
+export type StateListener<T> = (state: T) => void
+export type StateSelector<T, U> = (state: T) => U
+export type PartialState<T> = Partial<T> | ((state: T) => Partial<T>)
+
+export type State = Record<string, any>
+export type SetState<T> = (partialState: PartialState<T>) => void
+export type GetState<T> = () => T
+
+export type UseStore<T> = <U = T>(selector?: StateSelector<T, U>) => U
+
+export interface StoreApi<T> {
+  getState: GetState<T>
+  setState: SetState<T>
+  subscribe: (listener: StateListener<T>) => () => void
+  destroy: () => void
+}
 
 const reducer = <T>(state: any, newState: T) => newState
 
 export default function create<
-  State extends Record<string, any>,
-  SetState extends (partialState: PartialState<Record<string, any>>) => void,
-  GetState extends () => Record<string, any>
->(createState: (set: SetState, get: GetState) => State) {
-  const listeners: Set<StateListener<State>> = new Set()
+  TState extends State,
+  TSetState extends SetState<TState> = SetState<TState>,
```

Comment:
These `TSetState` and `TGetState` generics should just be removed. I was trying to use them to generate correct function signatures of the `set` and `get` functions passed into `createState`. I couldn’t get it to work though so we can simplify this bit by removing them:
```ts
export type CreateState<T> = (set: SetState<State>, get: GetState<State>) => T
...
export default function create<TState>(
  createState: CreateState<TState>
): [UseStore<TState>, StoreApi<TState>] {
```